File: python/programmers/stack_queue/기능개발.py
import math

def solution(progresses, speeds):
    answer = []
    needs = []
    for i in range(len(progresses)):
        progress = 100 - progresses[i]
        need = math.ceil(progress/speeds[i])
        needs.append(need)
    
    complete = 0
    most = 0
    for i in range(len(needs)):
        if complete == 0:
            complete += 1
            most = needs[i]
        elif needs[i] <= most:
            complete += 1
        else:
            answer.append(complete)
            complete = 1
            most = needs[i]
        if i == len(needs) - 1:
            answer.append(complete)
    return answer

File: java/src/main/java/ps_traning/programmers/level_2/기능개발.java
package ps_traning.programmers.level_2;

import java.util.ArrayList;
import java.util.LinkedList;
import java.util.List;
import java.util.Queue;


/**
 * 나눗셈에서 소수점 활용하려면 double로 형변환 해서 사용해야함.
 * Integer 상태에서 나누면 값도 Integer임.
 */
public class 기능개발 {
    public List<Integer> solution(int[] progresses, int[] speeds) {
        List<Integer> answer = new ArrayList<>();
        Queue<Integer> queue = new LinkedList<>();
        for (int i = 0; i < progresses.length; i++) {
            int need = (int) Math.ceil((double) (100 - progresses[i]) / speeds[i]);
            if (queue.isEmpty() || queue.peek() >= need) {
                queue.offer(need);
                continue;
            }
            answer.add(queue.size());
            queue.clear();
            queue.offer(need);
        }
        if (!queue.isEmpty()) {
            answer.add(queue.size());
        }
        return answer;
    }
}
